deltacat/compute/compactor_v2/steps/merge.py

```diff
@@ -168,32 +242,38 @@ def _download_compacted_table(
     hb_index: int,
     rcf: RoundCompletionInfo,
     read_kwargs_provider: Optional[ReadKwargsProvider] = None,
+    deletes_to_apply_by_stream_positions: Optional[pa.Table] = None,
     deltacat_storage=unimplemented_deltacat_storage,
     deltacat_storage_kwargs: Optional[dict] = None,
 ) -> pa.Table:
     tables = []
     hb_index_to_indices = rcf.hb_index_to_entry_range
-
     if str(hb_index) not in hb_index_to_indices:
         return None
-
     indices = hb_index_to_indices[str(hb_index)]
-
     assert (
         indices is not None and len(indices) == 2
     ), "indices should not be none and contains exactly two elements"
-
     for offset in range(indices[1] - indices[0]):
         table = deltacat_storage.download_delta_manifest_entry(
             rcf.compacted_delta_locator,
             entry_index=(indices[0] + offset),
             file_reader_kwargs_provider=read_kwargs_provider,
             **deltacat_storage_kwargs,
         )
-
         tables.append(table)
-
-    return pa.concat_tables(tables)
+    compacted_table = pa.concat_tables(tables)
+    if deletes_to_apply_by_stream_positions:
+        for delete_table in _get_all_deletes(deletes_to_apply_by_stream_positions):
+            compacted_table, number_of_rows_dropped = drop_rows_affected_by_deletes(
```

Comment:
Could we separate the `drop_rows_affected_by_deletes` apart from `_download_compacted_table` to another function like `drop_rows_from_compacted_table` so that it's clearer we are deleting from compacted table right after we download it?

Reply:
Good idea - addressing this in the next commit